technic_chests/init.lua

```diff
@@ -1,77 +1,74 @@
 
 technic = rawget(_G, "technic") or {}
 technic.chests = {}
 
-dofile(modpath.."/common.lua")
-dofile(modpath.."/register.lua")
-dofile(modpath.."/iron_chest.lua")
-dofile(modpath.."/copper_chest.lua")
-dofile(modpath.."/silver_chest.lua")
-dofile(modpath.."/gold_chest.lua")
-dofile(modpath.."/mithril_chest.lua")
+technic.chests.colors = {
+	{"black", S("Black")},
+	{"blue", S("Blue")},
+	{"brown", S("Brown")},
+	{"cyan", S("Cyan")},
+	{"dark_green", S("Dark Green")},
+	{"dark_grey", S("Dark Grey")},
+	{"green", S("Green")},
+	{"grey", S("Grey")},
+	{"magenta", S("Magenta")},
+	{"orange", S("Orange")},
+	{"pink", S("Pink")},
+	{"red", S("Red")},
+	{"violet", S("Violet")},
+	{"white", S("White")},
+	{"yellow", S("Yellow")},
+}
 
--- undo all of the locked wooden chest recipes created by default and
--- moreblocks, and just make them use a padlock.
+function technic.chests.change_allowed(pos, player, owned, protected)
+	if owned then
+		if minetest.is_player(player) and not default.can_interact_with_node(player, pos) then
+			return false
+		end
+	elseif protected then
+		if minetest.is_protected(pos, player:get_player_name()) then
+			return false
+		end
+	end
+	return true
+end
 
-if minetest.get_modpath("moreblocks") then
-	minetest.clear_craft({
-		type = "shapeless",
-		recipe = {
-			"default:chest",
-			"default:gold_ingot",
-		}
-	})
+if minetest.get_modpath("digilines") then
+	dofile(modpath.."/digilines.lua")
+end
 
-	minetest.clear_craft({
-		type = "shapeless",
+dofile(modpath.."/formspec.lua")
+dofile(modpath.."/inventory.lua")
+dofile(modpath.."/register.lua")
+dofile(modpath.."/chests.lua")
+
+-- Undo all of the locked wooden chest recipes, and just make them use a padlock.
+minetest.register_on_mods_loaded(function()
+	minetest.clear_craft({output = "default:chest_locked"})
+	minetest.register_craft({
+		output = "default:chest_locked",
 		recipe = {
-			"default:chest",
-			"default:bronze_ingot",
+			{ "group:wood", "group:wood", "group:wood" },
+			{ "group:wood", "basic_materials:padlock", "group:wood" },
+			{ "group:wood", "group:wood", "group:wood" }
 		}
 	})
-
-	minetest.clear_craft({
+	minetest.register_craft({
+		output = "default:chest_locked",
 		type = "shapeless",
 		recipe = {
 			"default:chest",
-			"default:copper_ingot",
+			"basic_materials:padlock"
 		}
 	})
-end
-
-minetest.clear_craft({
-	type = "shapeless",
-	recipe = {
-		"default:chest",
-		"default:steel_ingot",
-	}
-})
-
-minetest.clear_craft({output = "default:chest_locked"})
-
-minetest.register_craft({
-	output = "default:chest_locked",
-	recipe = {
-		{ "group:wood", "group:wood", "group:wood" },
-		{ "group:wood", "basic_materials:padlock", "group:wood" },
-		{ "group:wood", "group:wood", "group:wood" }
-	}
-})
-
-minetest.register_craft({
-	output = "default:chest_locked",
-	type = "shapeless",
-	recipe = {
-		"default:chest",
-		"basic_materials:padlock"
-	}
-})
+end)
 
+-- FIXME: Why does this LBM exist?
```

Comment:
I think that was for compatibility reasons a long time ago.
The default chest uses a `on_rightclick` sent formspec: https://github.com/minetest/minetest_game/blob/master/mods/default/chests.lua#L210
I guess it wasn't always like this.

**IMO**: remove it if you want, if it turns out to be an issue it should be added in the `default` mod anyway

Reply:
I think I'll remove it, I agree that it doesn't belong here in the first place.